src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2074,13 +2146,30 @@ private void onApplicationWindow(
                 initialBudget += credit;
                 initialPadding = padding;
 
-                subscription.onSubscribeSucceeded(traceId, authorization, packetId, ackIndex);
+                if (!isSubscribed)
+                {
+                    isSubscribed = true;
+                    subscription.onSubscribeSucceeded(traceId, authorization, packetId, ackIndex);
+                }
+
+                if (initialSlot != NO_SLOT)
+                {
+                    final MutableDirectBuffer buffer = bufferPool.buffer(initialSlot);
+                    final int offset = 0;
+                    final int limit = initialSlotOffset;
+
+                    flushApplicationData(initialSlotTraceId, authorization, buffer, offset, limit);
+                }
 
-                if (!MqttState.initialClosed(padding))
+                if (initialSlot == NO_SLOT)
                 {
-                    if (MqttState.initialClosing(state))
+                    if (!MqttState.initialClosed(padding))
```

Comment:
If `initial` is `closed` then we must have already sent an `ABORT`, `END` or received `RESET`, so then the `reaktor` will not call us with `WINDOW`, so no need to defend this case?